tests/unit/deploy/test_deployer.py

```diff
@@ -603,14 +604,70 @@ def test_lambda_deployer_initial_deploy(app_policy, sample_app):
         timeout=120, memory_size=256
     )
 
+class TestValidateCORS(object):
+    def test_cant_have_options_with_cors(self, sample_app):
+        @sample_app.route('/badcors', methods=['GET', 'OPTIONS'], cors=True)
+        def badview():
+            pass
+
+        with pytest.raises(ValueError):
+            validate_routes(sample_app.routes)
+
+    def test_cant_have_differing_cors_configurations(self, sample_app):
+        custom_cors = CORSConfig(
+            allow_origin='https://foo.example.com',
+            allow_headers=['X-Special-Header'],
+            max_age=600,
+            expose_headers=['X-Special-Header'],
+            allow_credentials=True
+        )
 
-def test_cant_have_options_with_cors(sample_app):
-    @sample_app.route('/badcors', methods=['GET', 'OPTIONS'], cors=True)
-    def badview():
-        pass
-
-    with pytest.raises(ValueError):
-        validate_routes(sample_app.routes)
+        @sample_app.route('/cors', methods=['GET'], cors=True)
+        def cors():
+            pass
+
+        @sample_app.route('/cors', methods=['PUT'], cors=custom_cors)
+        def different_cors():
+            pass
+
+        with pytest.raises(ValueError):
+            validate_routes(sample_app.routes)
+
+    def test_can_have_same_cors_configurations(self, sample_app):
```

Comment:
It would be nice to have another test like this one that explicitly tests custom CORSConfig objects that are `__eq__`.